### DISCUSSION TITLE
Embed Remix app into existing PHP app

### DISCUSSION THREAD

**Discussion by emipc** [remix-run/remix#4447] [Q&A] [answered]
Hi. In our company we have a huge product built in PHP and we are embedding CSR React apps in some pages.

My question is if would be possible in some way to embed a Remix app instead of our CSR ones.

The only way I can think of is by using an iframe, but I have had bad experiences in the past with them.

Any thoughts?

**Comment by sergiodxa** [comment, marked as answer]
Because Remix is not client-side only, you will either have to render an iframe pointing to your Remix app or make some routes be sent to Remix and the rest to your PHP app.

**Comment by machour** [comment]
@emipc I've been working on migrating a customer website made with Yii to Remix.

Using nginx, I'm able to serve some parts of the application with Remix, while the rest of the request gets routed to php-fpm.

Some bits:

`website.conf`
```conf
location /my-account {
    include /my/website/vhost/remix-proxy.conf;
}

location /build {
    include /my/website/vhost/remix-proxy.conf;
}

location /entry.worker.js {
    include /my/website/vhost/remix-proxy.conf;
}
```

`remix-proxy.conf`
```
proxy_pass http://127.0.0.1:3080;
proxy_http_version 1.1;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection 'upgrade';
proxy_set_header Host $host;
proxy_cache_bypass $http_upgrade;
```

I also rolled out my own `<Link />` & `<NavLink />` components to seemingly `reloadDocument` if I'm leaving a Remix page to go back to the Yii app:

```jsx
import { Link as RemixLink } from "@remix-run/react";
import { RemixLinkProps } from "@remix-run/react/dist/components";

export function Link(props: RemixLinkProps & React.RefAttributes<HTMLAnchorElement>) {
    const to = props.to as string;
    const reloadDocument = props.reloadDocument || !to.includes("/my-account");
    return <RemixLink {...props} reloadDocument={reloadDocument} />;
}
```

Give that a try and let me know if it helps!
